containers/board/SelectBox/index.tsx

```diff
@@ -0,0 +1,76 @@
+import styles from './index.module.scss';
+import Image from 'next/image';
+import { IconArrowDown, IconSort } from '@/public';
+import { MouseEvent, useState } from 'react';
+import Button from '@/components/Button';
+import {
+  SelectdValue,
+  selectList as defaultSelectList,
+} from '@/types/select-order.d';
+
+interface Props {
+  handleSelect: (selectedValue: SelectdValue) => void;
+  selectList: typeof defaultSelectList;
+}
+
+function SelectBox({ handleSelect, selectList = defaultSelectList }: Props) {
+  const [isListOpen, setIsListOpen] = useState(false);
+  const [selectedText, setSelectedText] = useState<SelectdValue>(selectList[0]);
+
+  const handleOpenList = () => {
+    if (isListOpen) {
+      setIsListOpen(() => false);
+    } else {
+      setIsListOpen(() => true);
+    }
+  };
+
+  const handleSelectItem = (e: MouseEvent<HTMLButtonElement>) => {
+    const target = e.target as HTMLButtonElement;
+    const textContent = target.textContent as SelectdValue;
+
+    if (textContent !== null) {
+      handleSelect(textContent);
+      setSelectedText(() => textContent);
+    } else {
+      console.warn('textContent is null');
+    }
+  };
```

Comment:
리액트에서 이벤트 객체를 다룰 때 조금 귀찮은 부분들이 있는 것 같습니다.
타입 단언 등을 어쩔수없이 사용하거나 오히려 타입에 맞춰 구현부의 로직을 변경하는 경우도 있구요. 🤔
이런 경우를 피하기 위해 `<Button>` 컴포넌트에서 명시적인 인자를 전달하는 방법도 있으니 참고해 보시면 좋을 것 같습니다!

Reply:
만약, 명시적인 인자를 활용할 경우 함수의 로직은 아래와 같이 단순화 할 수 있습니다. 😁
```suggestion
  const handleSelectItem = (orderType: SelectdValue) => {
    handleSelect(orderType);
    setSelectedText(() => orderType);
  };
```

Reply:
작성 예시는 `<Button>` 컴포넌트의 `onClick` props에 작성해 두었습니다!